eclipse-theia/theia
# Generic hover service support

#### What it does

Fixes https://github.com/eclipse-theia/theia/issues/11594
Fixes https://github.com/eclipse-theia/theia/issues/10867
Also addresses a comment I've left in another PR https://github.com/eclipse-theia/theia/pull/11083#discussion_r873871011.

Uses the `StatusBarHoverManager` (now `HoverService`) as a base for building a generic hover service for markdown/string/html display as hovers/tooltips. This retires the `TooltipService` (introduced in https://github.com/eclipse-theia/theia/pull/10108), which was the source of the error messages in https://github.com/eclipse-theia/theia/issues/11594.

One feature that is missing here is that the hover could appear at the cursor position, right now it's bound to the position of the target element. I plan to add that in a separate PR later on.

#### How to test

1. Confirm that hovers still appear correctly on status bars (see also https://github.com/eclipse-theia/theia/pull/11083)
2. Confirm that hovers appear correctly on icons of widgets on the side bar panels (navigator, debug, extension, preferences, etc.)
3. Confirm that hovers appear correctly on tree nodes (see also https://github.com/eclipse-theia/theia/pull/11708)
4. Confirm that hovers appear correctly in the vsx extension view (see also https://github.com/eclipse-theia/theia/pull/10108)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
